File: backend/app/models/transcode_preset.py
from sqlalchemy import Column, Integer, String, Boolean, Float, JSON, DateTime, func
from app.database import Base


class TranscodePreset(Base):
    __tablename__ = "transcode_presets"

    id = Column(Integer, primary_key=True, autoincrement=True)
    name = Column(String(100), nullable=False)
    description = Column(String(500), nullable=True)
    is_builtin = Column(Boolean, default=False)
    video_codec = Column(String(50), default="libx265")
    target_resolution = Column(String(20), nullable=True)
    bitrate_mode = Column(String(10), default="crf")
    crf_value = Column(Integer, nullable=True, default=23)
    target_bitrate = Column(String(20), nullable=True)
    hw_accel = Column(String(50), nullable=True)
    audio_mode = Column(String(20), default="copy")
    audio_codec = Column(String(50), nullable=True)
    container = Column(String(10), default="mkv")
    subtitle_mode = Column(String(20), default="copy")
    custom_flags = Column(String(1000), nullable=True)
    hdr_mode = Column(String(20), default="preserve")
    two_pass = Column(Boolean, default=False)
    encoder_tune = Column(String(50), nullable=True)
    created_at = Column(DateTime, server_default=func.now())
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())

File: frontend/MediaFlow/MediaFlow/Models/TranscodePreset.swift
import Foundation

struct TranscodePreset: Identifiable, Codable, Hashable {
    let id: Int
    let name: String
    var description: String?
    var isBuiltin: Bool = false
    var videoCodec: String = "libx265"
    var targetResolution: String?
    var bitrateMode: String = "crf"
    var crfValue: Int?
    var targetBitrate: String?
    var hwAccel: String?
    var audioMode: String = "copy"
    var audioCodec: String?
    var container: String = "mkv"
    var subtitleMode: String = "copy"
    var customFlags: String?
    var hdrMode: String = "preserve"
    var twoPass: Bool = false
    var encoderTune: String?

    var iconName: String {
        switch name {
        case "Balanced": return "slider.horizontal.3"
        case "Storage Saver": return "arrow.down.doc"
        case "Mobile Optimized": return "iphone"
        case "Ultra Fidelity": return "star"
        default: return "gearshape"
        }
    }
}
